File: 2019/day8/day8.py
f = open('input', 'r')

width = 25
height = 6
size = width * height
layers = f.read()

x = []

for i in range(len(layers)//size):
    cur = i*size
    x += [layers[cur:cur+size]]

for k in x:
    if not len(k) == size:
        print(len(k))

l = size
res = -1
for i in range(len(x)):
    if x[i].count('0') < l:
        l = x[i].count('0')
        res = i

print(x[res].count('1')*x[res].count('2'))

image = ['2']*size

for layer in x:
    for i in range(size):
        if image[i] == '2':
            image[i] = layer[i]

for i in range(height):
    cur = i*width
    print(image[cur:cur+width])

File: 2019/day8/day8.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

int main() {
    std::string input;
    std::cin >> input;

    const int width = 25;
    const int height = 6;
    const int size = width * height;

    std::vector<std::string> layers;
    for (int i = 0; i < input.size() / size; i++) {
        int cur = i * size;
        layers.push_back(input.substr(cur, size));
    }

    int min = size;
    int res = -1;
    for (int i = 0; i < layers.size(); i++) {
        int occ = std::count(layers[i].begin(), layers[i].end(), '0');
        if (occ < min) {
            min = occ;
            res = i;
        }
    }

    int prod = std::count(layers[res].begin(), layers[res].end(), '1') * std::count(layers[res].begin(), layers[res].end(), '2');
    printf("%d\n", prod);

    std::vector<char> img(size);
    std::fill(img.begin(), img.end(), '2');
    for (const auto& layer : layers) {
        for (int i = 0; i < size; i++) {
            if (img[i] == '2') {
                switch (layer[i])
                {
                    case '0':
                        img[i] = ' ';
                        break;
                    case '1':
                        img[i] = 'x';
                        break;
                }
            }
        }
    }

    std::string image(img.data());

    for (int i = 0; i < height; i++) {
        int cur = i * width;
        printf("%s\n", image.substr(cur, width).c_str());
    }
}